lib/OperationHandler.cpp

```diff
@@ -435,31 +455,39 @@ ScanResponse OperationHandler::HandleScanRequestSerialized(const ScanRequest& sc
         scanRequest->ssid_len > 0 ? std::make_optional<const Ssid>(gsl::span{scanRequest->ssid, scanRequest->ssid_len}) : std::nullopt;
 
     // Start all scan requests
-    std::vector<std::future<std::vector<ScannedBss>>> futureScanResults;
+    using FutureScanResult = std::future<std::pair<std::vector<ScannedBss>, ScanStatus>>;
+    std::vector<std::pair<GUID, FutureScanResult>> futureScanResults;
     for (auto& wlanIntf : m_wlanInterfaces)
     {
         try
         {
-            futureScanResults.push_back(wlanIntf->Scan(requestedSsid));
+            futureScanResults.emplace_back(wlanIntf->GetGuid(), wlanIntf->Scan(requestedSsid));
```

Comment:
in the cases there are >1 interfaces, do we need to block on running scans serially across each interface before processing any scan results? e.g. once we get the first scan results, we can process them. then process the scan results from the next interface?

Reply:
Scans are running concurrently: we schedule them all first (getting a `std::future` that is not completed at the time), and then wait until all the futures are completed.
The time spent waiting for results here is only as long as the time taken by the slowest scan among all the interfaces.